pkg/constants/constants.go

```diff
@@ -313,10 +313,10 @@ func ISOLoaderPath(arch string) string {
 
 // ISOKernelPath returns path use to store the kernel
 func ISOKernelPath(arch string) string {
-	return ISOLoaderPath(arch) + "linux"
+	return filepath.Join(ISOLoaderPath(arch), "linux")
```

Comment:
@ldevulder note the arch is passed as an argument instead of using runtime architecture to enable cross arch builds. We might be looking to build arm64 images in a x86_64 host, I think we are so close to make that possible.